app/scripts/migrations/127.ts

```diff
@@ -0,0 +1,453 @@
+      rpcPrefs: { blockExplorerUrl: 'https://sepolia.etherscan.io' },
+    },
+    {
+      type: 'infura',
+      id: 'linea-sepolia',
+      chainId: '0xe705',
+      ticker: 'LineaETH',
+      nickname: LINEA_SEPOLIA_DISPLAY_NAME,
+      rpcUrl: 'https://linea-sepolia.infura.io/v3/{infuraProjectId}',
+      rpcPrefs: { blockExplorerUrl: 'https://sepolia.lineascan.build' },
+    },
+    {
+      type: 'infura',
+      id: 'linea-mainnet',
+      chainId: '0xe708',
+      ticker: 'ETH',
+      nickname: LINEA_MAINNET_DISPLAY_NAME,
+      rpcUrl: 'https://linea-mainnet.infura.io/v3/{infuraProjectId}',
+      rpcPrefs: { blockExplorerUrl: 'https://lineascan.build' },
+    },
+    ...networkConfigurations,
+  ];
+
+  // Group the network configurations by by chain id, producing
+  // a mapping from chain id to an array of network configurations
+  const networkConfigurationArraysByChainId = networkConfigurations.reduce(
+    (acc: Record<string, RuntimeObject[]>, networkConfiguration) => {
+      if (
+        isObject(networkConfiguration) &&
+        typeof networkConfiguration.chainId === 'string'
+      ) {
+        (acc[networkConfiguration.chainId] ??= []).push(networkConfiguration);
+      }
+      return acc;
+    },
+    {},
+  );
+
+  // Get transaction history in reverse chronological order to help with tie breaks
+  const transactions: RuntimeObject[] = Array.isArray(
+    state.TransactionController.transactions,
+  )
+    ? state.TransactionController.transactions
+        .filter(
+          (tx) =>
+            isObject(tx) &&
+            typeof tx.time === 'number' &&
+            typeof tx.networkClientId === 'string',
+        )
+        .sort((a, b) => b.time - a.time)
+    : [];
+
+  // For each chain id, merge the array of network configurations
+  const networkConfigurationsByChainId = Object.entries(
+    networkConfigurationArraysByChainId,
+  ).reduce((acc: Record<string, unknown>, [chainId, networks]) => {
+    //
+    // Calculate the tie breaker network, whose values will be preferred
+    let tieBreaker: RuntimeObject | undefined;
+
+    // If one of the networks is the globally selected network, use that
+    tieBreaker = networks.find(
+      (network) => network.id === networkState.selectedNetworkClientId,
+    );
+
+    // Otherwise use the network that was most recently transacted on
+    if (!tieBreaker) {
+      transactions
+        .filter((tx) => tx.chainId === chainId)
+        .some(
+          (tx) =>
+            (tieBreaker = networks.find(
+              (network) => network.id === tx.networkClientId,
+            )),
+        );
+    }
+
+    // If no transactions were found for the chain id,
+    // try to fall back to the built in infura network
+    if (!tieBreaker) {
+      tieBreaker = networks.find((network) => network.type === 'infura');
+    }
+
+    // Calculate the unique set of valid rpc endpoints for this chain id
+    const rpcEndpoints = networks.reduce(
```

Comment:
The code to calculate `rpcEndpoints` is noticably easier to read as of 74f01a33458

I'm not sure if it is because I am reading it after a good nights sleep, but I think the reduction in `if` blocks and the removal of the function defined in line helps. Also the explicit reuse of `networkConfigurations` for searching for duplicates, and the naming of a variable as `existingEndpoint` helps. And the code comment before the `.Object.values(acc).flatMap((n)` line clarified something for me as well.

So nice work!